src/julia.h

```diff
@@ -179,6 +179,20 @@ STATIC_INLINE int jl_array_ndimwords(uint32_t ndims)
     return (ndims < 3 ? 0 : ndims-2);
 }
 
+typedef struct {
+    union {
```

Comment:
the GC won't see the pointer (or the bytevec), in general, if it is declared immutable. putting it inside another object (such as an ASCIIString) is sufficient to hide it from GC tracking.